tests/test_config.py

```diff
@@ -549,3 +583,130 @@ def test_s3_url_different_models_create_different_directories(mock_pull_files):
+        compilation_config=vllm_config.compilation_config,
+        model_config=model_config,
+    )
+
+    # Verify general defaults
+    for k, v in default_config["general"].items():
+        if k == "pass_config":
+            for pass_k, pass_v in default_config["general"]["pass_config"].items():
+                actual = getattr(vllm_config.compilation_config.pass_config, pass_k)
+                expected = resolve_config_value(pass_v, vllm_config)
+                assert actual == expected
+        else:
+            actual = getattr(vllm_config.compilation_config, k)
+            expected = resolve_config_value(v, vllm_config)
+            assert actual == expected
+
+    # Verify quantized-specific defaults
+    for k, v in default_config["is_quantized"]["pass_config"].items():
+        actual = getattr(vllm_config.compilation_config.pass_config, k)
+        expected = resolve_config_value(v, vllm_config)
+        assert actual == expected
+
+    # Verify sequential-specific defaults
+    for k, v in default_config["is_sequential"]["pass_config"].items():
+        actual = getattr(vllm_config.compilation_config.pass_config, k)
+        expected = resolve_config_value(v, vllm_config)
+        assert actual == expected
+
+
+def test_vllm_config_callable_defaults():
+    """Test that callable defaults work in the config system.
+
+    Verifies that lambdas in default configs can inspect VllmConfig properties
+    (e.g., is_quantized, is_model_moe) to conditionally set optimization flags.
+    """
+    # Test resolve_config_value with different callable scenarios
+    config_no_model = VllmConfig(optimization_level=OptimizationLevel.O2)
+
+    # Callable that checks if model exists
+    has_model = lambda cfg: cfg.model_config is not None
+    assert resolve_config_value(has_model, config_no_model) is False
+    assert resolve_config_value(True, config_no_model) is True
+    assert resolve_config_value(False, config_no_model) is False
+
+    # Test with quantized model
+    quantized_model = ModelConfig("jerryzh168/Qwen3-8B-INT4")
+    config_quantized = VllmConfig(
+        model_config=quantized_model, optimization_level=OptimizationLevel.O2
+    )
+    enable_if_quantized = lambda cfg: (
+        cfg.model_config is not None and cfg.model_config.is_quantized()
+    )
+    assert resolve_config_value(enable_if_quantized, config_quantized) is True
+    assert resolve_config_value(enable_if_quantized, config_no_model) is False
+
+    # Test with MoE model
+    moe_model = ModelConfig("deepseek-ai/DeepSeek-V2-Lite")
+    config_moe = VllmConfig(
+        model_config=moe_model, optimization_level=OptimizationLevel.O2
+    )
+    enable_if_sequential = lambda cfg: (
+        cfg.model_config is not None and not cfg.model_config.is_model_moe()
+    )
+    assert resolve_config_value(enable_if_sequential, config_moe) is False
+    assert resolve_config_value(enable_if_sequential, config_quantized) is True
```

Comment:
Could we get another test that creates a bunch of different VllmConfig instances and verifies that different properties are set the way we expect them?